From 5e2acc4d36b613e1d65e0aae9c290dae1be15c79 Mon Sep 17 00:00:00 2001
From: Carlos Alberto Lopez Perez <clopez@igalia.com>
Date: Tue, 8 Nov 2016 00:33:26 +0100
Subject: [PATCH] Using trap SIG${X} in scripts is not POSIX compliant (bug
 #1705)

 * Stop also trying to trap SIGKILL as it is not possible.
---
 autocfg                     | 2 +-
 test/test-airdecap-ng.sh    | 2 +-
 test/test-airolib-sqlite.sh | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/autocfg b/autocfg
index fadaf3b..9cbcaf9 100644
--- a/autocfg
+++ b/autocfg
@@ -43,7 +43,7 @@ if [ "$(uname -s)" = 'OpenBSD' ]; then
 	trap clean_exit EXIT
 else
 	OPENBSD=0
-	trap clean_exit SIGINT SIGKILL SIGQUIT SIGSEGV SIGPIPE SIGALRM SIGTERM EXIT
+	trap clean_exit INT QUIT SEGV PIPE ALRM TERM EXIT
 fi
 
 # $1 flag $2 variable
diff --git a/test/test-airdecap-ng.sh b/test/test-airdecap-ng.sh
index 070a88c..5018323 100755
--- a/test/test-airdecap-ng.sh
+++ b/test/test-airdecap-ng.sh
@@ -27,7 +27,7 @@ compute_sha1() {
 if [ "$(uname -s)" = 'OpenBSD' ]; then
 	trap "rm -rf "${tmpdir}"" EXIT
 else
-	trap "rm -fr "${tmpdir}"" SIGINT SIGKILL SIGQUIT SIGSEGV SIGPIPE SIGALRM SIGTERM EXIT
+	trap "rm -fr "${tmpdir}"" INT QUIT SEGV PIPE ALRM TERM EXIT
 fi
 # Test1
 cp -f "${TESTDIR}/wpa.cap" "${tmpdir}"
diff --git a/test/test-airolib-sqlite.sh b/test/test-airolib-sqlite.sh
index 5e09d82..a8e81f7 100755
--- a/test/test-airolib-sqlite.sh
+++ b/test/test-airolib-sqlite.sh
@@ -5,7 +5,7 @@
 TESTDIR="$(dirname $0)"
 tmpfile="$(mktemp -u -t acng.XXXX)"
 # Clean on exit
-trap "rm -fr "${tmpdir}"" SIGINT SIGKILL SIGQUIT SIGSEGV SIGPIPE SIGALRM SIGTERM EXIT
+trap "rm -fr "${tmpdir}"" INT QUIT SEGV PIPE ALRM TERM EXIT
 
 echo Harkonen | ./airolib-ng "${tmpfile}" --import essid -
 [ $? -ne 0 ] && exit 1
-- 
2.1.4

